components/homme/src/share/cxx/ExecSpaceDefs.hpp

```diff
@@ -383,11 +383,19 @@ struct Dispatch<HommexxGPU> {
     // serialize parallel scans.
 
     // Detect the value type
+#ifdef HOMMEXX_ENABLE_GPU
+    using value_type =
+      typename Kokkos::Impl::FunctorAnalysis
+        < Kokkos::Impl::FunctorPatternInterface::SCAN
+        , void
+        , Lambda, void >::value_type ;
+#else
     using value_type =
       typename Kokkos::Impl::FunctorAnalysis
         < Kokkos::Impl::FunctorPatternInterface::SCAN
         , void
         , Lambda >::value_type ;
```

Comment:
So just adding the extra `void` template argument doesn't work for the non-GPU case, thus requiring the `#ifdef`?